src/services/projects.js

```diff
@@ -318,4 +321,199 @@ module.exports = class ProjectsHelper {
 			throw error
 		}
 	}
+
+	static async submitForReview(resourceId, bodyData, userDetails) {
+		try {
+			let projectDetails = await this.details(resourceId, userDetails.organization_id, userDetails.id)
+
+			if (projectDetails.statusCode !== httpStatusCode.ok) {
+				return responses.failureResponse({
+					message: 'DONT_HAVE_PROJECT_ACCESS',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			let projectData = projectDetails.result
+
+			if (projectData.user_id !== userDetails.id) {
+				return responses.failureResponse({
+					message: 'DONT_HAVE_PROJECT_ACCESS',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			let entityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
+				{
+					model: common.PROJECT,
+					status: common.STATUS_ACTIVE,
+				},
+				userDetails,
+				['id', 'value', 'has_entities', 'validations']
+			)
+
+			// Using forEach instead of for loop for entity type validations
+
+			entityTypes.forEach((entityType) => {
+				const fieldData = projectData[entityType.value]
+
+				if (entityType.validations.required) {
+					let required = utils.checkRequired(entityType, fieldData)
+					if (!required) {
+						throw responses.failureResponse({
+							message: `${entityType.value} not added`,
+							statusCode: httpStatusCode.bad_request,
+							responseCode: 'CLIENT_ERROR',
+							error: utils.errorObject(common.BODY, entityType.value),
+						})
+					}
+				}
+
+				if (entityType.has_entities) {
+					let checkEntities = utils.checkEntities(entityType, fieldData)
```

Comment:
Where do we check single-select vs multi-select?

Reply:
in utils file we are checking and what ever we are getting we are validating @aks30